connectors/webhook/src/main/java/io/camunda/connector/inbound/authorization/AuthorizationHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.inbound.authorization;
+
+import io.camunda.connector.api.inbound.webhook.WebhookProcessingPayload;
+import io.camunda.connector.inbound.authorization.AuthorizationHandler.NoOpAuthorizationHandler;
+import io.camunda.connector.inbound.model.WebhookAuthorization;
+import io.camunda.connector.inbound.model.WebhookAuthorization.None;
+
+public abstract sealed class AuthorizationHandler<T extends WebhookAuthorization>
+    permits ApiKeyAuthHandler, NoOpAuthorizationHandler, BasicAuthHandler, JWTAuthHandler {
+  protected T expectedAuthorization;
+  protected WebhookProcessingPayload payload;
+
+  public AuthorizationHandler(T authorization, WebhookProcessingPayload payload) {
+    this.expectedAuthorization = authorization;
+    this.payload = payload;
+  }
+
+  /**
+   * Check if authorization is present in the payload. If false, the webhook endpoint returns 401.
+   */
+  public abstract boolean isPresent();
+
+  /** Check if authorization is valid. If false, the webhook endpoint returns 403. */
+  public abstract boolean isValid();
+
+  public static NoOpAuthorizationHandler noOp() {
+    return NoOpAuthorizationHandler.INSTANCE;
+  }
```

Comment:
What about a stateless interface to void object creation per request?

Presence and validation result could be modeled via an `AuthorizationResult` sealed class hierarchy. For example:
*  AuthorizationResult.Successful
* AuthorizationResult.Failed
  * AuthorizationResult.Failed.InvalidCredentials
  * AuthorizationResult.Failed.Unauthorized
...